src/pyodide-py/_pyodide/docstring.py

```diff
@@ -0,0 +1,31 @@
+from textwrap import dedent
+
+
+def dedent_docstring(docstring):
+    first_newline = docstring.find("\n")
+    if first_newline == -1:
+        return docstring
+    docstring = docstring[:first_newline] + dedent(docstring[first_newline:])
+    return docstring
+
+
+def get_cmeth_docstring(func):
+    """Get the value to use for the PyMethodDef.ml_doc attribute for a builtin
+    function. This is used in docstring.c.
+
+    The ml_doc should start with a signature which cannot have any type
+    annotations. The signature must end with the exact characters ")\n--\n\n".
+    For example: "funcname(arg1, arg2)\n--\n\n"
```

Comment:
Maybe we could do a doctest here for some builtin function that's unlikely to change such as "sum" ?